sklearnex/neighbors/_lof.py

```diff
@@ -0,0 +1,180 @@
+        }
+
+    def __init__(
+        self,
+        n_neighbors=20,
+        *,
+        algorithm="auto",
+        leaf_size=30,
+        metric="minkowski",
+        p=2,
+        metric_params=None,
+        contamination="auto",
+        novelty=False,
+        n_jobs=None,
+    ):
+        super().__init__(
+            n_neighbors=n_neighbors,
+            algorithm=algorithm,
+            leaf_size=leaf_size,
+            metric=metric,
+            p=p,
+            metric_params=metric_params,
+            n_jobs=n_jobs,
+            contamination=contamination,
+            novelty=novelty,
+        )
+
+    @run_with_n_jobs
+    def _onedal_fit(self, X, y, queue=None):
+        if sklearn_check_version("1.2"):
+            self._validate_params()
+
+        onedal_params = {
+            "n_neighbors": self.n_neighbors,
+            "algorithm": self.algorithm,
+            "metric": self.effective_metric_,
+            "p": self.effective_metric_params_["p"],
+        }
+
+        try:
+            requires_y = self._get_tags()["requires_y"]
+        except KeyError:
+            requires_y = False
+
+        self._onedal_estimator = onedal_NearestNeighbors(**onedal_params)
+        self._onedal_estimator.requires_y = requires_y
+        self._onedal_estimator.effective_metric_ = self.effective_metric_
+        self._onedal_estimator.effective_metric_params_ = self.effective_metric_params_
+        self._onedal_estimator.fit(X, y, queue=queue)
+
+        self._save_attributes()
+
+        if self.contamination != "auto":
+            if not (0.0 < self.contamination <= 0.5):
+                raise ValueError(
+                    "contamination must be in (0, 0.5], " "got: %f" % self.contamination
+                )
+
+        n_samples = self.n_samples_fit_
+
+        if self.n_neighbors > n_samples:
+            warnings.warn(
+                "n_neighbors (%s) is greater than the "
+                "total number of samples (%s). n_neighbors "
+                "will be set to (n_samples - 1) for estimation."
+                % (self.n_neighbors, n_samples)
+            )
+        self.n_neighbors_ = max(1, min(self.n_neighbors, n_samples - 1))
+
+        self._distances_fit_X_, _neighbors_indices_fit_X_ = self.kneighbors(
+            n_neighbors=self.n_neighbors_
+        )
+
+        # Sklearn includes a check for float32 at this point which may not be
+        # necessary for onedal
+
+        self._lrd = self._local_reachability_density(
+            self._distances_fit_X_, _neighbors_indices_fit_X_
+        )
+
+        # Compute lof score over training samples to define offset_:
+        lrd_ratios_array = self._lrd[_neighbors_indices_fit_X_] / self._lrd[:, np.newaxis]
+
+        self.negative_outlier_factor_ = -np.mean(lrd_ratios_array, axis=1)
+
+        if self.contamination == "auto":
+            # inliers score around -1 (the higher, the less abnormal).
+            self.offset_ = -1.5

```

Comment:
Looks very unintuitive.can it be let say -2.0?

Reply:
Direct from sklearn unfortunately